server/src/main/java/org/opensearch/cluster/routing/allocation/decider/DiskThresholdDecider.java

```diff
@@ -208,6 +215,8 @@ public Decision canAllocate(ShardRouting shardRouting, RoutingNode node, Routing
                 );
             }
             return Decision.YES;
+        } else if (REMOTE_CAPABLE.equals(getShardPool(shardRouting, allocation))) {
```

Comment:
Would this if-else be a bit simpler to follow if structured like the following?

```java
if (REMOTE_CAPABLE.equals(getShardPool(shardRouting, allocation)) {
    if (REMOTE_CAPABLE.equals(getNodePool(node))) {
        ...
    } else {
        return Decision.NO;
    }
}
```

I _think_ that is logically equivalent. Returning "NO" in that else block is much easier to follow in my opinion. It's really a style question (and does result in indenting the code more), so I'll defer to your opinion.

Reply:
Another option:

```java
if (REMOTE_CAPABLE.equals(getShardPool(shardRouting, allocation)) {
    if (REMOTE_CAPABLE.equals(getNodePool(node)) == false) {
        return Decision.NO;
    }
    ...
}
```

Reply:
I like the second option

Reply:
I suddenly realized that we may not be able to do this, because if a node has both data role and search role and when we try to allocate local storage index, this logic will be executed. I tend not to change this piece of code.